cmd/add.go

```diff
@@ -216,12 +246,36 @@ func init() {
 		"pretend to be the this user (only works if you started the server)")
 	addCmd.Flags().StringVar(&setMetadata, "metadata", "",
 		"key=val;key=val metadata to apply to all files in the set")
+	addCmd.Flags().Var(&setReason, "reason",
+		"storage reason: 'backup' | 'archive' | 'quarantine'")
+	addCmd.Flags().StringVar(&setReview, "review", "",
+		"months/years until review date, provided in format: <number><unit>, e.g. 1y for 1 year")
+	addCmd.Flags().StringVar(&setRemoval, "remove", "",
+		"months/years until removal date, provided in format: <number><unit>, e.g. 1y for 1 year")
 
 	if err := addCmd.MarkFlagRequired("name"); err != nil {
 		die(err.Error())
 	}
 }
 
+func (r *Reason) Set(value string) error {
+	if slices.Contains(Reasons, value) {
+		*r = Reason(slices.Index(Reasons, value))
+
+		return nil
+	}
+
+	return ErrInvalidReason
+}
+
+func (r *Reason) String() string {
+	return Reasons[*r]
```

Comment:
If you make this method on a non-pointer (`func (r Reason) String() string {`), you can avoid the deref here.